includes/functions/llms-functions-l10n.php

```diff
@@ -0,0 +1,160 @@
+
+	$plugin_dir   = $plugin_dir ? $plugin_dir : LLMS_PLUGIN_DIR;
+	$language_dir = $language_dir ? $language_dir : 'languages';
+
+	/**
+	 * Load from the custom LifterLMS "safe" directory (if it exists).
+	 *
+	 * Example path: wp-content/languages/lifterlms/lifterlms-en_US.mo
+	 */
+	load_textdomain( $domain, sprintf( '%1$s/%2$s-%3$s.mo', llms_l10n_get_safe_directory(), $domain, llms_get_locale( $domain ) ) );
+
+	/**
+	 * Load from default plugin locations specified by the WP core.
+	 *
+	 * 1. wp-content/languages/plugins/lifterlms-en_US.mo
+	 * 2. wp-content/plugins/lifterlms/languages/lifterlms-en_US.mo
+	 */
+	load_plugin_textdomain( $domain, false, sprintf( '%1$s/%2$s', basename( $plugin_dir ), $language_dir ) );
+
+}
+
+/**
+ * Load JSON format localization files for a script
+ *
+ * This function localizes using the WP Core's default language file directories as a after
+ * checking in the LifterLMS-defined "safe" directory.
+ *
+ * Language files files can be found in the following locations (The first loaded file takes priority):
+ *
+ *   1. wp-content/languages/{$domain}/{$domain}-{$locale}-{md5($handle)}.json
+ *
+ *      This is recommended "safe" location where custom language files can be stored. A file
+ *      stored in this directory will never be automatically overwritten.
+ *
+ *   2. wp-content/languages/plugins/{$domain}-{$locale}-{md5($handle)}.json
+ *
+ *      This is the default directory where WordPress will download language files from the
+ *      WordPress GlotPress server during updates. If you store a custom language file in this
+ *      directory it will be overwritten during updates.
+ *
+ *   3. wp-content/plugins/{$domain}/languages/{$domain}-{$locale}-{md5($handle)}.json
+ *
+ *      This is the the LifterLMS plugin directory. A language file stored in this directory will
+ *      be removed from the server during a LifterLMS plugin update.
+ *
+ * @since [version]
+ *
+ * @param string      $handle       Handle for the script. The script must be registered with `wp_register_script()` for this to work!
+ * @param string      $domain       Textdomain being loaded.
+ * @param string|null $plugin_dir   Full path to the plugin directory, if none supplied `LLMS_PLUGIN_DIR` is used.
+ * @param string|null $language_dir Relative path to the language directory within the plugin. If none supplied, `languages` is used.
+ * @return void
+ */
+function llms_set_script_translations( $handle, $domain, $plugin_dir = null, $language_dir = null ) {
+
+	$plugin_dir   = $plugin_dir ? $plugin_dir : LLMS_PLUGIN_DIR;
+	$language_dir = $language_dir ? $language_dir : 'languages';
+
+	$safe_dir = llms_l10n_get_safe_directory();
+
+	// Check if the file exists in the safe directory (#1) and load it from there if it does, otherwise fall back to the default directories (#2 then #3).
+	$path = load_script_textdomain( $handle, $domain, $safe_dir ) ? $safe_dir : sprintf( '%1$s/%2$s', $plugin_dir, $language_dir );
```

Comment:
I'm not sure I get this...
who falls back on #2 then #3?

Reply:
After writing tests for this method I realized that while it technically works it also technically doesn't. 

See the rewrite (and included tests)